tests/integration/endpoint_test.go

```diff
@@ -14,17 +14,33 @@ func endpointTests(t *testing.T, when spec.G, it spec.S) {
 	var service testutil.TestService
 	var stagedService testutil.TestService
 
+	it.Before(func() {
+		service.Create(t, "ibuildthecloud/demo:v1")
+		stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
+	})
 	it.After(func() {
 		service.Remove()
 		stagedService.Remove()
 	})
 
 	when("a service is running and another is staged", func() {
 		it("should have endpoints that are available on both", func() {
-			service.Create(t, "ibuildthecloud/demo:v1")
-			stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
 			assert.Equal(t, "Hello World", service.GetEndpoint())
 			assert.Equal(t, "Hello World v3", stagedService.GetEndpoint())
+			kubeUrlv0 := service.GetKubeEndpointUrlByRevision("v0")
+			kubeUrlv3 := stagedService.GetKubeEndpointUrlByRevision("v3")
+			rioUrlv0 := service.GetEndpointUrl()
+			rioUrlv3 := stagedService.GetEndpointUrl()
+			assert.Equal(t, kubeUrlv0, rioUrlv0)
+			assert.Equal(t, kubeUrlv3, rioUrlv3)
+
+			assert.Equal(t, service.GetEndpointResponse(kubeUrlv0), service.GetEndpointResponse(rioUrlv0))
```

Comment:
This whole block and associated methods aren't intuitive anymore, and there is some overlap in functionality across methods. What if we do: 
```
testUtil.WaitForURLResponse(service.GetEndpointURL) == "Hello World"
testUtil.WaitForURLResponse(stagedService.GetEndpointURL) == "Hello World V3"
service.GetEndpointURL == service.GetKubeEndpointURL
stagedService.GetEndpointURL == stagedService.GetKubeEndpointURL
```
Then for global test below lets do same: 
```
testUtil.WaitForURLResponse(service.GetAppEndpointURL) == "Hello World"
service.GetAppEndpointURL == service.GetKubeAppEndpointURL
```
That way we only have 2 real methods: endpointURL and appEndpointURL, and the equivalent kube versions of them. Might be contradicting myself, but we don't need to do a check on KubeEndpoint response if the URLs are equivalent.